Mods/DiscordRichPresence/Src/DiscordClient.cpp

```diff
@@ -0,0 +1,28 @@
+#include "DiscordClient.h"
+#include "Logging.h"
+
+static const char* APPLICATION_ID = "852754886197379103";
+
+void DiscordClient::Initialize()
+{
+	DiscordEventHandlers handlers;
+	memset(&handlers, 0, sizeof(handlers));
+	Discord_Initialize(APPLICATION_ID, &handlers, 1, NULL);
+}
+
+void DiscordClient::Update(const char* state, const char* details, const char* imageKey)
+{
+	DiscordRichPresence discordPresence;
+	memset(&discordPresence, 0, sizeof(discordPresence));
+	discordPresence.state = state;
+	discordPresence.details = details;
+	discordPresence.largeImageKey = imageKey;
```

Comment:
These string pointers will no longer be valid after `OnLoadScene` goes out of scope. Is the Discord SDK ok with that or does it expect for the data to be long lived?